kgforge/specializations/models/rdf/service.py

```diff
@@ -162,18 +181,20 @@ def validate(self, resource: Resource, type_: str):
                     "Resource has list of types as attribute and type_ parameter is not specified. "
                     "Please provide a type_ parameter to validate against it."
                 )
-            if type_ is None:
-                shape_iri = self.types_to_shapes[resource.type]
-            else:
-                shape_iri = self.types_to_shapes[type_]
+            type_to_validate = type_ if type_ else resource.type
+            shape_iri = self.get_shape_iri_from_class_fragment(type_to_validate)
         except AttributeError as exc:
-            raise TypeError("Resource requires a type attribute") from exc
-
+            raise TypeError(
+                "Resource requires a type attribute or a type_ parameter should be provided"
+            ) from exc
         data_graph = as_graph(resource, False, self.context, None, None)
```

Comment:
this line assumes that the resource context is the same as the given context, while this is not the case when loading the model from the directory and validating a resource with a context from a store

Reply:
Yes. The service is not making clear that self.context should be the one used for resources to validate. I recommend to have that in a follow up PR